input/witnessgen.go

```diff
@@ -262,5 +266,103 @@ func (wt WitnessType) GenWitnessFunc(signer Signer,
 			return nil, fmt.Errorf("unknown witness type: %v", wt)
 		}
 	}
+}
+
+// GetWitnessSizeUpperBound returns the maximum length of the witness for
+// the given type if it would be included in a tx. We also return if the
+// output itself is a nested p2sh output, if so then we need to take into
+// account the extra sigScript data size.
+func GetWitnessSizeUpperBound(wt WitnessType) (int, bool, error) {
+	switch wt {
+
+	// Outputs on a remote commitment transaction that pay directly to us.
+	case CommitSpendNoDelayTweakless:
```

Comment:
Isn't this switch an indication that `GetWitnessSizeUpperBound` should really be a method of `WitnessType`? Or having a `WitnessType` interface, so that witness types can be implemented independently?